### ISSUE TITLE
Not working in TBC

### ISSUE THREAD

**Issue by bondjw07** [darfink/ItemAutocomplete#4]
This addon was absolutly amazing in classic.  I was using it on a daily basis.  I was hoping we might be able to get this working in TBC as well.  Right now it is throwing the following error when typing anything after '['.

`Message: ...emAutocomplete\Source\ChatAutocompleteIntegrator.lua:136: attempt to call method 'ClearAll' (a nil value)`
`Time: Wed May 26 19:29:50 2021`
`Count: 4`
`Stack: ...emAutocomplete\Source\ChatAutocompleteIntegrator.lua:136: attempt to call method 'ClearAll' (a nil value)`
`Locals: `

Happy to do some testing or provide more information if that is helpful.

**Comment by darfink** [comment]
There's actually an [active pull request](https://github.com/darfink/ItemAutocomplete/pull/3) for TBC support, but I've been waiting on some additional input before merging it. But I may just merge it and do the final touch ups myself. The problem remains though, that I do not have TBC: Classic installed. If I could provide you with an experimental TBC build, perhaps you'd be able to confirm whether it's working as intended or not?

**Comment by darfink** [comment]
I've uploaded an experimental TBC build, it would be awesome if you are able to confirm whether it works or not:
https://transfer.sh/1W1DVV2/iac-tbc.zip

**Comment by akay** [comment]
Seems to work pretty well for me! I wish it also included spells and abilities too though, now that we can link them :) But then it wouldn't be called ItemAutocomplete, I guess!

**Comment by boktai1000** [comment]
@darfink I've been using the pull request and I would say it works pretty much without issue but I did manage to make a Lua error happen today by typing [ then copy pasting an item name in, and like backspacing or something and hitting enter or something.

Message: ...emAutocomplete\Source\ChatAutocompleteIntegrator.lua:134: attempt to perform arithmetic on a string value
Time: Sun May 30 23:44:33 2021
Count: 2
Stack: ...emAutocomplete\Source\ChatAutocompleteIntegrator.lua:134: attempt to perform arithmetic on a string value
[string "@Interface\AddOns\ItemAutocomplete\Source\ChatAutocompleteIntegrator.lua"]:134: in function `_OnItemSearchComplete'
[string "@Interface\AddOns\ItemAutocomplete\Source\ChatAutocompleteIntegrator.lua"]:183: in function `onFinish'
[string "@Interface\AddOns\ItemAutocomplete\Source\Shared\TaskScheduler.lua"]:77: in function `_OnUpdate'
[string "@Interface\AddOns\ItemAutocomplete\Source\Shared\TaskScheduler.lua"]:19: in function <...Ons\ItemAutocomplete\Source\Shared\TaskScheduler.lua:19>

Locals: self = <table> {
 itemLinkDelimiters = <table> {
 }
 itemDatabase = <table> {
 }
 editBoxCursorOffsets = <table> {
 }
 original = <table> {
 }
 buttonMenu = <unnamed> {
 }
 methods = <table> {
 }
}
editBox = ChatFrame1EditBox {
 0 = <userdata>
 tabCompleteTableIndex = 1
 autoCompleteSource = <function> defined =[C]:-1
 setText = 0
 customAutoCompleteFunction = <function> defined @Interface\FrameXML\ChatFrame.lua:4036
 chatFrame = ChatFrame1 {
 }
 addSpaceToAutoComplete = true
 language = "Common"
 disallowAutoComplete = false
 headerSuffix = ChatFrame1EditBoxHeaderSuffix {
 }
 text = ""
 autoCompleteXOffset = 35
 header = ChatFrame1EditBoxHeader {
 }
 addHighlightedText = true
 languageID = 7
 autoCompleteParams = <table> {
 }
}
items = <function> defined @Interface\AddOns\ItemAutocomplete\Source\ItemDatabase.lua:211
searchInfo = <table> {
 searchTerm = "Enchant Bracer - Lesser Strength"
 searchOffsetX = 0
}
searchTerm = nil
(*temporary) = <function> defined @Interface\AddOns\ItemAutocomplete\Source\Utility\Functions.lua:27
(*temporary) = nil
(*temporary) = true
(*temporary) = ""
(*temporary) = ""
(*temporary) = nil
(*temporary) = 0
(*temporary) = nil
(*temporary) = "attempt to perform arithmetic on a string value"
util = <table> {
}

**Comment by darfink** [comment]
Yes, I believe that problem you experienced is only exhibited in the pull request.
It stems from this line (which is not part of the latest release):
```lua
util.PrettyPrint('Item search DISCARDED' + searchTerm)
```
So your issue can be resolved by using the ZIP referenced in this thread instead.

As a PSA I may mention that the I intend to update ItemAutocomplete wtih TBC: Classic support on CurseForge, but since its merger with Overwolf, I can no longer access my account... And their support have yet to resolve the problem.

**Comment by boktai1000** [comment]
The only other issue I have run into when using ItemAutocomplete was with using it with Decursive in the past - I am now using hte latest 2.0 version uploaded to Curseforge - saw the site was down and you must have gotten access back to your account

Otherwise the only other question I have is if these changes will also make it to Classic Era - personally, I plan on playing BC Classic full time but plan on dabling with Classic Era.

Thanks for the update and great addon! Your work is appreciated!

**Comment by darfink** [comment]
Yes, I was actually suggested a potential fix for this just now, and I have an implementation locally that I am about to test out. This alternative solution would resolve the existing issue with Decursive. With that said, if there's interest for it, I'll probably backport these changes to WoW: Classic as well.

**Comment by boktai1000** [comment]
@darfink Yeah! That sounds good, I see your commits to the repo and have been watching closely. Looking forward to it!

**Comment by darfink** [comment]
Alright, I've added "duo-support" so the same AddOn is compatible with Classic & BCC simultaneously.
I've tested v2.0.1 in both classic & BCC, and from my basic testing in Classic it now seems compatible with both Prat & Decursive.

Please give it a try: https://github.com/darfink/ItemAutocomplete/releases/tag/v2.0.1

**Comment by darfink** [comment]
g
